### PR TITLE
feat: update campaign function

### PR DESCRIPTION
**Pull request by 0xged**

### CODE REVIEW

File: solidity/contracts/OngoingAirdrops.sol [hunk 1]
```diff
@@ -37,7 +37,39 @@ contract OngoingAirdrops is AccessControl, IOngoingAirdrops {
     bytes32 _campaign,
     bytes32 _root,
     TokenAmount[] calldata _tokensAllocation
-  ) external override {}
+  ) external override onlyRole(ADMIN_ROLE) {
+    if (_campaign == bytes32(0)) revert InvalidCampaign();
+    if (_root == bytes32(0)) revert InvalidMerkleRoot();
+    if (_tokensAllocation.length == 0) revert InvalidTokenAmount();
+
+    for (uint256 i = 0; i < _tokensAllocation.length; i++) {
```
Comment (nchamo) [inline solidity/contracts/OngoingAirdrops.sol:45]: Is it cheaper if we don't initialize i?

File: solidity/contracts/OngoingAirdrops.sol [hunk 2]
```diff
@@ -37,7 +37,39 @@ contract OngoingAirdrops is AccessControl, IOngoingAirdrops {
...
+      // Build our unique ID for campaign and token address.
```
Comment (nchamo) [inline solidity/contracts/OngoingAirdrops.sol:46]: It's usually cheaper if you load the TokenAmount struct into memory and only read the array once

File: test/unit/ongoing-airdrops.spec.ts [hunk 3]
```diff
@@ -1,34 +1,46 @@
...
+      tokens.push(await smock.fake('IERC20'));
+    }
     ongoingAirdrops = await ongoingAirdropsFactory.deploy(superAdmin.address, [admin.address]);
     [superAdminRole, adminRole] = await Promise.all([ongoingAirdrops.SUPER_ADMIN_ROLE(), ongoingAirdrops.ADMIN_ROLE()]);
     snapshot = await takeSnapshot();
   });
 
   beforeEach(async () => {
     await snapshot.restore();
+    for (let i = 0; i < 10; i++) {
```
Comment (nchamo) [inline test/unit/ongoing-airdrops.spec.ts:40]:
```suggestion
    for (let i = 0; i < tokens.length; i++) {
```

**Review by nchamo — APPROVED** [review]
Great work! Just suggested some optimizations

File: solidity/contracts/OngoingAirdrops.sol (hunk 1 above)
Comment (0xged) [reply solidity/contracts/OngoingAirdrops.sol:45]: As said on TG. We had done previous tests that showed not assigning ` = 0` was an optimization... But new tests show it doesn't change. As always, I'm confused about the optimizer.

File: solidity/contracts/OngoingAirdrops.sol (hunk 2 above)
Comment (0xged) [reply solidity/contracts/OngoingAirdrops.sol:46]: I was completely skeptic about this, I really thought `calldata` access was cheaper that `memory`. Wrong, once again 😬 .

**Review by nchamo — APPROVED** [review]

**Pull request merged**